src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -951,14 +1015,26 @@ ss::future<cluster::init_tm_tx_reply> tx_gateway_frontend::init_tm_tx(
         leader_opt = _leaders.local().get_leader(tx_ntp);
     }
     if (!leader_opt) {
-        vlog(txlog.warn, "can't find {} in the leaders cache", tx_ntp);
+        vlog(
+          txlog.warn,
+          "init_tm_tx request failed, tx_id {}, can't find {} in the leaders "
+          "cache",
+          tx_id,
+          tx_ntp);
         co_return cluster::init_tm_tx_reply{tx_errc::leader_not_found};
     }
 
     auto leader = leader_opt.value();
     auto _self = _controller->self();
 
     if (leader != _self) {
+        vlog(
+          txlog.trace,
+          "init_tm_tx request failed, tx_id {}, no leader match, leader: {}, "
+          "self: {}",
```

Comment:
nit: maybe we can explicitly output that current node is not a leader (coordinator) for given partition